ledger-tool/src/main.rs

```diff
@@ -628,40 +628,50 @@ fn main() {
 
     let ledger_path = PathBuf::from(value_t_or_exit!(matches, "ledger", String));
 
-    let genesis_config = GenesisConfig::load(&ledger_path).unwrap_or_else(|err| {
-        eprintln!(
-            "Failed to open ledger genesis_config at {:?}: {}",
-            ledger_path, err
-        );
-        exit(1);
-    });
+    let genesis_config = || {
+        GenesisConfig::load(&ledger_path).unwrap_or_else(|err| {
+            eprintln!(
+                "Failed to open ledger genesis_config at {:?}: {}",
+                ledger_path, err
+            );
+            exit(1);
+        })
+    };
 
-    let blocktree = match Blocktree::open(&ledger_path) {
+    let blocktree = || match Blocktree::open(&ledger_path) {
```

Comment:
Too cute. Now a noun is being used to describe a function that returns a noun with the same name.  How about just moving this code into top-level `create_blocktree` and `create_database` functions?

Reply:
Could  `open_blocktree` or something similar be ok since `create` could imply actual on-disk creation (versus loading/opening, which this is)?

Reply:
Yep, sounds good, thanks